app/views/applications/_pet_info.html.erb

```diff
@@ -0,0 +1,19 @@
+<% @application.find_pets.each do |pet| %>
+  <div id="<%= pet.id %>">
+    <% if pet.find_application_pet(@application.id).status == "Pending" && local_assigns[:admin] %>
+      <%= pet.name %>
+      <%= button_to "Approve Application", "/admin/applications/#{@application.id}/application_pets/#{pet.find_application_pet(@application.id).id}?status=approve", method: :patch %>
+      <%= button_to "Reject Application", "/admin/applications/#{@application.id}/application_pets/#{pet.find_application_pet(@application.id).id}?status=reject", method: :patch %>
```

Comment:
I left a comment in your `routes.rb` file but I will mention it again, to make your button routes more restful, they should be using this path: `"/admin/applications/#{@application.id}/pets/#{pet.id}?status=approve"`.

This way you can avoid having to call `pet.find_application_pet(@application.id).id`.

One other suggestion is another way to pass the status back to the controller without having to use query params.

```erb
<%= button_to "Approve Application", "/admin/applications/#{@application.id}/pets/#{pet.id}", method: :patch, params: { status: approve } %>
```